File: sdks/csharp/sdk/src/PogodocApi/Templates/Requests/UpdateTemplateRequest.cs
using System.Text.Json.Serialization;
using PogodocApi.Core;

namespace PogodocApi;

[Serializable]
public record UpdateTemplateRequest
{
    [JsonPropertyName("templateInfo")]
    public required UpdateTemplateRequestTemplateInfo TemplateInfo { get; set; }

    [JsonPropertyName("previewIds")]
    public required UpdateTemplateRequestPreviewIds PreviewIds { get; set; }

    /// <summary>
    /// ID by which the new template content is saved
    /// </summary>
    [JsonPropertyName("contentId")]
    public required string ContentId { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/php/src/client/src/Templates/Requests/UpdateTemplateRequest.php
<?php

namespace Pogodoc\Templates\Requests;

use Pogodoc\Core\Json\JsonSerializableType;
use Pogodoc\Templates\Types\UpdateTemplateRequestTemplateInfo;
use Pogodoc\Core\Json\JsonProperty;
use Pogodoc\Templates\Types\UpdateTemplateRequestPreviewIds;

class UpdateTemplateRequest extends JsonSerializableType
{
    /**
     * @var UpdateTemplateRequestTemplateInfo $templateInfo
     */
    #[JsonProperty('templateInfo')]
    public UpdateTemplateRequestTemplateInfo $templateInfo;

    /**
     * @var UpdateTemplateRequestPreviewIds $previewIds
     */
    #[JsonProperty('previewIds')]
    public UpdateTemplateRequestPreviewIds $previewIds;

    /**
     * @var string $contentId ID by which the new template content is saved
     */
    #[JsonProperty('contentId')]
    public string $contentId;

    /**
     * @param array{
     *   templateInfo: UpdateTemplateRequestTemplateInfo,
     *   previewIds: UpdateTemplateRequestPreviewIds,
     *   contentId: string,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->templateInfo = $values['templateInfo'];
        $this->previewIds = $values['previewIds'];
        $this->contentId = $values['contentId'];
    }
}
